libs/components/forms/src/lib/modules/checkbox/checkbox.component.ts

```diff
@@ -46,48 +47,62 @@ export class SkyCheckboxComponent implements ControlValueAccessor, OnInit {
    * If the checkbox does not include a visible label, use `label` instead.
    */
   @Input()
-  public labelledBy: string;
+  public labelledBy: string | undefined;
 
   /**
    * Specifies an ID for the checkbox.
    * @default a unique, auto-incrementing integer. For example: `sky-checkbox-1`
    */
+  @HostBinding('attr.id')
   @Input()
-  public id = `sky-checkbox-${++nextId}`;
+  public set id(value: string | undefined) {
+    if (value) {
+      this.#_id = value;
+      this.inputId = `input-sky-checkbox-${value}`;
+    } else {
+      const defaultId = `sky-checkbox-${++nextId}`;
```

Comment:
Do we want to change this to a new ID every time a falsy value is specified? I'd prefer to create a default ID in the constructor and then set it back to that ID each time. Ideally we'd only ever `++nextId` once per component so that all the other IDs we generate for the component instance contain that same value.

Reply:
Done